goal_src/jak2/levels/common/airlock.gc

```diff
@@ -7,3 +7,1762 @@
+                                                      (set! s4-0 (-> ctywide borrow-level 1))
+                                                      )
+                                                  (set! s3-0 (cdr s3-0))
+                                                  (set! s2-0 (car s3-0))
+                                                  )
+                )
+          (when s4-0
+            (let ((v1-17 (level-status *level* s4-0)))
+              (if (not (or (= v1-17 'loaded) (= v1-17 'active)))
+                  (return #f)
+                  )
+              )
+            )
+          (if (!= arg0 'display)
+              (return #t)
+              )
+          (let ((a1-7 (car gp-1)))
+            (while (not (null? gp-1))
+              (let ((v1-26 (level-get *level* (the-as symbol a1-7))))
+                (if (not (and v1-26 (= (-> v1-26 status) 'active) (-> v1-26 display?)))
+                    (return #f)
+                    )
+                )
+              (set! gp-1 (cdr gp-1))
+              (set! a1-7 (car gp-1))
+              )
+            )
+          #t
+          )
+        (else
+          (let* ((v1-33 gp-1) (a0-11 (car v1-33))) (while (not (null? v1-33))
+                                                     (dotimes (a1-10 6)
+                                                       (if (= a0-11 (-> *load-state* want a1-10 name))
+                                                           (goto cfg-50)
+                                                           )
+                                                       )
+                                                     #t
+                                                     (return #f)
+                                                     (label cfg-50)
+                                                     (set! v1-33 (cdr v1-33))
+                                                     (set! a0-11 (car v1-33))
+                                                     )
+                )
+          #t
+          )
+        )
+      )
+    )
+  )
+
+(defmethod rotate-gear! com-airlock ((obj com-airlock) (arg0 float))
+  (when (nonzero? (-> obj gear))
+    (if (and (zero? (-> obj gear-sound-id))
+             (-> obj sound-gear)
+             (and (-> obj next-state) (= (-> obj next-state name) 'open))
+             (>= (com-airlock-method-25 obj (target-pos 0) #f) 0.0)
+             )
+        (set! (-> obj gear-sound-id) (sound-play-by-spec (-> obj sound-gear) (new-sound-id) (the-as vector #t)))
+        )
+    (seek! (-> obj gear-rotv) arg0 (* 131072.0 (-> self clock seconds-per-frame)))
+    (+! (-> obj gear-rot) (* (-> obj gear-rotv) (-> self clock seconds-per-frame)))
+    (twist-set! (-> obj gear) (the-as float #f) (the-as float #f) (-> obj gear-rot))
+    )
+  (-> obj gear-rotv)
+  )
+
+;; WARN: Return type mismatch int vs none.
+(defmethod com-airlock-method-27 com-airlock ((obj com-airlock) (arg0 symbol))
```

Comment:
`play-city-voice-sound` maybe? I think this plays the "welcome back" sound clip.